src/elf.rs

```diff
@@ -687,24 +714,22 @@ impl<'a> Elf<'a> {
     }
 }
 
-impl<'a> Object for Elf<'a> {
-    fn to_bytes(artifact: &Artifact) -> Result<Vec<u8>, Error> {
-        // TODO: make new fully construct the elf object, e.g., the definitions, imports, and links don't take self
-        // this means that a call to new has a fully constructed object ready to marshal into bytes, similar to the mach backend
-        let mut elf = Elf::new(&artifact);
-        for def in artifact.definitions() {
-            debug!("Def: {:?}", def);
-            elf.add_definition(def.name, def.data, def.prop);
-        }
-        for (ref import, ref kind) in artifact.imports() {
-            debug!("Import: {:?} -> {:?}", import, kind);
-            elf.import(import.to_string(), kind);
-        }
-        for link in artifact.links() {
-            elf.link(&link);
-        }
-        let mut buffer = Cursor::new(Vec::new());
-        elf.write(&mut buffer)?;
-        Ok(buffer.into_inner())
+pub fn to_bytes(artifact: &Artifact) -> Result<Vec<u8>, Error> {
```

Comment:
Shouldn’t this be a method in Elf?

Similarly could leave object trait and have all the targets implement this method, along with emit ? Or perhaps not necessary, dunno